test/integration/typescript/pages/_error.tsx

```diff
@@ -0,0 +1,11 @@
+import { NextPageContext } from 'next'
+import ErrorComponent from 'next/error'
+
+class CustomError extends ErrorComponent {
+  static async getInitialProps({ res }: NextPageContext) {
```

Comment:
This function is also sync by nature. Might be worth reflecting this in some future PR.

Reply:
It's probably best to make it `async` so we have that flexibility in the future IMO.

Reply:
Making new PR for ti 😇